pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -202,17 +202,23 @@ func (r *ReconcileVirtualMachineImport) Reconcile(request reconcile.Request) (re
 		return reconcile.Result{}, err
 	}
 
+	// Init provider:
+	provider, err := r.createProvider(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if instance.DeletionTimestamp != nil && utils.HasFinalizer(instance, utils.RestoreVMFinalizer) {
+		r.finalize(instance, provider)
```

Comment:
We might enter the `r.restoreInitialVMState(vmiName, provider)` in this code path, where the provider wasn't initialized, and as a result to failure to connect to it.

We need to make sure we passed through `r.initProvider` before invoking the `r.finalize`

Reply:
Ack